apps/prairielearn/src/pages/instructorAssessmentQuestions/components/EditQuestionModal.tsx

```diff
@@ -0,0 +1,491 @@
+  const isInherited = (fieldName: keyof ZoneQuestionJson | keyof QuestionAlternativeJson) => {
+    if (!isAlternative) return false;
+    return (
+      (!(fieldName in localQuestion) ||
+        localQuestion[fieldName as keyof typeof localQuestion] === undefined) &&
+      fieldName in alternativeGroup &&
+      alternativeGroup[fieldName as keyof typeof alternativeGroup] !== undefined
+    );
+  };
+
+  // Check if points/autoPoints is inherited (special case since we track which one to use)
+  const isPointsInherited = () => {
+    if (!isAlternative) return false;
+    return (
+      localQuestion[originalPointsProperty] === undefined &&
+      alternativeGroup[originalPointsProperty] !== undefined
+    );
+  };
+
+  // Format auto points for display (handles both single numbers and arrays)
+  const formatAutoPoints = (value: number | number[] | undefined): string => {
+    if (value === undefined) return '';
+    if (Array.isArray(value)) return value.join(',');
+    return String(value);
+  };
+
+  // Parse auto points from string to array (only used on save, not on every keystroke)
+  const parseAutoPointsForSave = (value: string | number | number[]): number | number[] => {
+    // If it's already a number or array, return as-is
+    if (typeof value !== 'string') {
+      return value;
+    }
+
+    if (assessmentType === 'Exam') {
+      // For exams, parse string as array
+      const trimmed = value.trim();
+      if (trimmed === '') return [];
+      return trimmed
+        .split(',')
+        .map((v) => Number(v.trim()))
+        .filter((v) => !Number.isNaN(v));
+    }
+    // For homework, parse as single number
+    return Number(value);
+  };
+
+  // Get the auto points value to display (own or inherited)
+  // Uses the originalPointsProperty to determine which field to check
+  const getAutoPointsDisplayValue = () => {
+    const ownValue = localQuestion[originalPointsProperty];
+    if (ownValue !== undefined) {
+      return formatAutoPoints(ownValue);
+    }
+    if (isAlternative) {
+      const inheritedValue = alternativeGroup[originalPointsProperty];
+      if (inheritedValue !== undefined) {
+        return formatAutoPoints(inheritedValue);
+      }
+    }
+    return '';
+  };
+
+  useEffect(() => {
+    void Promise.resolve().then(() => {
+      const newQuestion = structuredClone(question);
+      setLocalQuestion(newQuestion);
+    });
+  }, [question]);
+
+  // Get the max auto points value to display (own or inherited)
+  // Uses the originalMaxProperty to determine which field to check
+  const getMaxAutoPointsDisplayValue = () => {
+    const ownValue = localQuestion[originalMaxProperty];
+    if (ownValue !== undefined) {
+      return ownValue;
+    }
+    if (isAlternative) {
+      const inheritedValue = alternativeGroup[originalMaxProperty];
+      if (inheritedValue !== undefined) {
+        return inheritedValue;
+      }
+    }
+    return '';
+  };
```

Comment:
Can you talk more about the choice to use functions here rather than just computing a `const maxAutoPointsDisplayValue = ...` variable?